api/server/request_controller.go

```diff
@@ -0,0 +1,117 @@
+package server
+
+import (
+	"net/http"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/gin-gonic/gin"
+	"github.com/iron-io/functions/api"
+	"github.com/iron-io/runner/common"
+)
+
+type RequestController interface {
+	AppName() string
+	SetAppName(string)
+	RoutePath() string
+	SetRoutePath(string)
+	Request() *http.Request
+	Response() http.ResponseWriter
+	Logger() logrus.FieldLogger
+	SetLogger(logrus.FieldLogger)
```

Comment:
I think we remove these, the logger is optional in the middleware and I don't think we want to lock ourselves into these functions on the interface forever (unless we're 100% sure, which we're not).